packages/block-editor/src/store/reducer.js

```diff
@@ -264,16 +281,34 @@ function withIgnoredBlockChange( reducer ) {
  * @return {Function} Enhanced reducer function.
  */
 const withInnerBlocksRemoveCascade = ( reducer ) => ( state, action ) => {
-	if ( state && action.type === 'REMOVE_BLOCKS' ) {
-		const clientIds = [ ...action.clientIds ];
+	const getAllChildren = ( clientIds ) => {
+		const result = [ ...clientIds ];
 
 		// For each removed client ID, include its inner blocks to remove,
 		// recursing into those so long as inner blocks exist.
-		for ( let i = 0; i < clientIds.length; i++ ) {
-			clientIds.push( ...state.order[ clientIds[ i ] ] );
+		for ( let i = 0; i < result.length; i++ ) {
+			if ( state.order[ result[ i ] ] ) {
```

Comment:
Is this condition related to your comment at https://github.com/WordPress/gutenberg/pull/16392#issuecomment-507735698 with regards to orphaned references? Should we comment it as a temporary fix if it is meant to be temporary?

Reply:
no, to be completely honest, I tried but failed to find why this condition is necessary. When you remove an "empty paragraph" using backspace, there's a JS error that is triggered if you remove this condition.